src/main/java/com/esri/samples/na/find_service_areas_for_multiple_facilities/FindServiceAreasForMultipleFacilitiesSample.java

```diff
@@ -0,0 +1,230 @@
+        mapView.addDrawStatusChangedListener(new DrawStatusChangedListener() {
+          @Override
+          public void drawStatusChanged(DrawStatusChangedEvent drawStatusChangedEvent) {
+            if (drawStatusChangedEvent.getDrawStatus() == DrawStatus.COMPLETED) {
+              // enable the 'find service areas' button
+              findServiceAreasButton.setDisable(false);
+              mapView.removeDrawStatusChangedListener(this);
+            }
+          }
+        });
+
+        // determine the service areas and display them when the button is clicked
+        findServiceAreasButton.setOnAction(event -> {
+
+          // disable the button
+          findServiceAreasButton.setDisable(true);
+
+          // show the progress indicator
+          progressIndicator.setVisible(true);
+
+          // create a service area task from URL
+          ServiceAreaTask serviceAreaTask = new ServiceAreaTask("https://sampleserver6.arcgisonline.com/arcgis/rest/services/NetworkAnalysis/SanDiego/NAServer/ServiceArea");
+          serviceAreaTask.loadAsync();
+
+          // create default service area task parameters
+          ListenableFuture<ServiceAreaParameters> serviceAreaTaskParametersFuture = serviceAreaTask.createDefaultParametersAsync();
+          serviceAreaTaskParametersFuture.addDoneListener(() -> {
+            try {
+              ServiceAreaParameters serviceAreaParameters = serviceAreaTaskParametersFuture.get();
+              // set the task parameters to have the task return polygons
+              serviceAreaParameters.setPolygonDetail(ServiceAreaPolygonDetail.HIGH);
+              serviceAreaParameters.setReturnPolygons(true);
+              // clear the default service area, and add service areas of 1 minute and 3 minutes travel time by car
+              serviceAreaParameters.getDefaultImpedanceCutoffs().clear();
+              serviceAreaParameters.getDefaultImpedanceCutoffs().addAll(Arrays.asList(1.0, 3.0));
+
+              // create query parameters used to select all facilities from the feature table
+              QueryParameters queryParameters = new QueryParameters();
+              queryParameters.setWhereClause("1=1");
+
+              // add all facilities to the service area parameters
+              serviceAreaParameters.setFacilities(facilitiesTable, queryParameters);
+
+              // find the service areas around the facilities using the parameters
+              ListenableFuture<ServiceAreaResult> serviceAreaResultFuture = serviceAreaTask.solveServiceAreaAsync(serviceAreaParameters);
+              serviceAreaResultFuture.addDoneListener(() -> {
+                try {
+                  // get the task results
+                  ServiceAreaResult serviceAreaResult = serviceAreaResultFuture.get();
+
+                  // display all the service areas that were found to the map view
+                  List<Graphic> serviceAreaGraphics = serviceAreasGraphicsOverlay.getGraphics();
```

Comment:
being pedantic, this line doesn't do what the comment suggests. The comment applies to the next block.

Reply:
Amended to `// create a list to hold all the found service areas`